nncf/tensor/tensor.py

```diff
@@ -65,7 +65,7 @@ def __bool__(self) -> bool:
         return bool(self.data)
 
     def __iter__(self) -> Iterator[Tensor]:
-        return TensorIterator(self.data)
+        return TensorIterator(self)
```

Comment:
Could you check if use nncf.Tensor without TensorIterator, 
Looks like it have same behavior 
```python
    def __iter__(self) -> Iterator[Tensor]:
        if not self.shape:
            raise TypeError("iteration over a 0-d tensor")
        return iter(self)
```